### PR TITLE
Diona wall opening

### PR DESCRIPTION
**Pull request by Geevies**
Diona can now open biomass walls by clicking them.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/random_map/automata/diona.dm [hunk 1]
```diff
@@ -1,6 +1,25 @@
 /turf/simulated/wall/diona/Initialize(mapload)
 	. = ..(mapload, "biomass")
 
+/turf/simulated/wall/diona/attack_hand(mob/user)
+	if(istype(user, /mob/living/carbon/alien/diona) || istype(user, /mob/living/carbon/human/diona))
```
Comment (Geevies) [inline code/modules/random_map/automata/diona.dm]: is_diona is fucking weird, so I didn't use it.

Comment (NonQueueingMatt) [reply code/modules/random_map/automata/diona.dm]: You should use it anyway. From what I can see it works fine.

Comment (Geevies) [inline code/modules/random_map/automata/diona.dm]:
```suggestion
	if(user.is_diona())
```

File: code/modules/random_map/automata/diona.dm [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+		attack_generic(user)
+	else
+		..()
+
+/turf/simulated/wall/diona/attack_generic(mob/user, var/damage, var/attack_message)
+	if(istype(user, /mob/living/carbon/alien/diona) || istype(user, /mob/living/carbon/human/diona))
```
Comment (Geevies) [inline code/modules/random_map/automata/diona.dm]:
```suggestion
	if(user.is_diona())
```

**Comment by Arrow768** [comment]
Imho that should not be possible of all walls as that would require some mapping gymnastics in case you want to map a diona wall that even they can't pass through.

Either add a specific subtype for that or add/change the diona-material-door to be usable exclusively by diona.

File: code/modules/random_map/automata/diona.dm [hunk 3]
```diff
@@ -1,6 +1,31 @@
...
+		..()
+
+/turf/simulated/wall/diona/attack_generic(mob/user, var/damage, var/attack_message)
+	if(user.is_diona() && acts_as_door)
+		if(can_open == WALL_OPENING)
+			return
+		can_open = WALL_CAN_OPEN
+		user.visible_message(span("notice", "\The [user] strokes its feelers against \the [src] and the biomass [density ? "moves aside" : "closes up"]."))
+		toggle_open(user)
+		sleep(15)
```
Comment (Arrow768) [inline code/modules/random_map/automata/diona.dm]: use a timer instead

Comment (Sindorman) [reply code/modules/random_map/automata/diona.dm]: agreed

**Comment by Arrow768** [comment]
I am also not too sure, if its a good idea to have that as the default, as it might break existing maps (i.e. the diona away mission if you have a diona with you)

**Pull request closed without merging**